go/vt/vtgate/vschema_manager_test.go

```diff
@@ -472,13 +472,44 @@ func TestMarkErrorIfCyclesInFk(t *testing.T) {
 						},
 					},
 				}
-				_ = vschema.AddForeignKey("ks", "t2", createFkDefinition([]string{"col"}, "t1", []string{"col"}, sqlparser.Cascade, sqlparser.Cascade))
-				_ = vschema.AddForeignKey("ks", "t3", createFkDefinition([]string{"col"}, "t2", []string{"col"}, sqlparser.Cascade, sqlparser.Cascade))
-				_ = vschema.AddForeignKey("ks", "t1", createFkDefinition([]string{"col"}, "t3", []string{"col"}, sqlparser.Cascade, sqlparser.Cascade))
+				_ = vschema.AddForeignKey("ks", "t2", createFkDefinition([]string{"col"}, "t1", []string{"col"}, sqlparser.SetNull, sqlparser.SetNull))
+				_ = vschema.AddForeignKey("ks", "t3", createFkDefinition([]string{"col"}, "t2", []string{"col"}, sqlparser.SetNull, sqlparser.SetNull))
+				_ = vschema.AddForeignKey("ks", "t1", createFkDefinition([]string{"col"}, "t3", []string{"col"}, sqlparser.SetNull, sqlparser.SetNull))
 				return vschema
 			},
 			errWanted: "VT09019: keyspace 'ks' has cyclic foreign keys",
```

Comment:
Do we want to change this error message to be more explicit? Like `cyclic cascade foreign key constraints` or something like that? How do we want to tell the user that specifically cascade is part of the problem and they can avoid it without a cascade?

Reply:
What do you propose? I could print the cycle we find as part of the error message, because Shlomi already added code to find the cycle as well.

Reply:
I've improved the error message to also have the cycle reported in it.